src/inmanta/server/services/compilerservice.py

```diff
@@ -650,7 +650,11 @@ async def request_recompile(
         requested = datetime.datetime.now().astimezone()
 
         shared_keys = mergeable_env_vars.keys() & env_vars.keys()
-        assert not shared_keys, f"An env var can not be both mergeable and normal: {shared_keys}"
+        if shared_keys:
+            raise Exception(
```

Comment:
```suggestion
            raise ValueError(
```

I think this would be slightly more to the point, but I'm fine with either.